src/imgtools/coretypes/box.py

```diff
@@ -0,0 +1,386 @@
+        """Expand the bounding box by a specified padding value in all directions.
+
+        Parameters
+        ----------
+        padding : int
+            The padding value to expand the bounding box.
+
+        Returns
+        -------
+        RegionBox
+            The expanded bounding box.
+        """
+        if padding == 0:
+            return self
+
+        padded_min = Coordinate3D(
+            self.min.x - padding,
+            self.min.y - padding,
+            self.min.z - padding,
+        )
+        padded_max = Coordinate3D(
+            self.max.x + padding,
+            self.max.y + padding,
+            self.max.z + padding,
+        )
+        return RegionBox(min=padded_min, max=padded_max)
+
+    def expand_to_cube(self, desired_size: int | None = None) -> RegionBox:
+        """Convert the bounding box to a cube by making the size equal along all dimensions.
+
+        This is done by finding which dimension is the largest,
+        and then pad the other dimensions to make them equal to the desired size.
+
+        Parameters
+        ----------
+        desired_size : int | None
+            The desired size of the cube. If None, the maximum dimension size is used
+
+        Returns
+        -------
+        RegionBox
+            The bounding box converted to a cube.
+
+        Raises
+        ------
+        ValueError
+            If the desired size is smaller than the current maximum dimension size.
+        """
+        max_size = max(self.size)
+
+        if not desired_size:
+            return self.minimum_dimension_size(max_size)
+
+        if desired_size < max_size:
+            msg = (
+                f"Desired size {desired_size} is smaller than"
+                f" the current maximum dimension size {max_size}."
+            )
+            raise ValueError(msg)
+
+        return self.minimum_dimension_size(desired_size)
+
+    def minimum_dimension_size(self, size: int = 5) -> RegionBox:
+        """Ensure that the bounding box has a minimum size along each dimension.
+
+        Parameters
+        ----------
+        size : int
+            The minimum size of the bounding box along each dimension.
+
+        Returns
+        -------
+        RegionBox
+            The bounding box with a minimum size along each dimension.
+
+        Notes
+        -----
+        Validation is done to ensure that any min coordinates that are negative are set to 0,
+        and the difference is added to the maximum coordinates
+        """
+        extra_x = max(0, size - self.size.width) // 2
+        extra_y = max(0, size - self.size.height) // 2
+        extra_z = max(0, size - self.size.depth) // 2
+
+        min_coord = self.min - (extra_x, extra_y, extra_z)
+        max_coord = self.max + (extra_x, extra_y, extra_z)
+
+        self._adjust_negatived_coordinates(min_coord, max_coord)
+
+        return RegionBox(min=min_coord, max=max_coord)
+
+    def _adjust_negatived_coordinates(
```

Comment:
I think this is just a typo?
```suggestion
    def _adjust_negative_coordinates(
```